proposals/0184-improved-pointers.md

```diff
@@ -309,6 +309,29 @@ while pixel < base + size
 }
 ```
 
+- **attempt to fully support partial initialization**
+
+Currently, in buffer pointers, sourced move memorystate operations like move-assignment and move-initialization assume that the source contains the same or fewer elements as the destination. In other words, it assumes the destination buffer is larger than the source buffer. This covers the common use case, since in Swift, collection buffers generally grow over time, not shrink. However, sometimes we want to do other things with our buffer pointers. For example:
+
+ - get a pointer to an element offset `n` strides from the beginning of the buffer
+ - copy part of a buffer pointer into another, smaller buffer pointer
+ - initialize or assign items at an offset from the beginning of the buffer
+
+Unfortunately, it is still not very ergonomic to do these things, even under this proposal, as they still require dropping down to `UnsafeMutablePointer` and `UnsafeMutableRawPointer`. 
+
+Right now, sourced-copy buffer operations like assignment and initialization attempt to cover the most general case, making no assumptions about the relative size of the two buffers. This proposal does not attempt to change that (flawed?) precendent. The return value of these methods is a tuple containing 
```

Comment:
no way to check before hand, but we basically call `it.next()!` for each slot in the destination buffer. Exhausting the sequence before the destination is full is a programming error. Conversely, supporting partially initialized buffers is done by turning that around; running out of room in the destination buffer is considered a programming error. However, the second condition can be checked with a `_debugPrecondition` beforehand.